cluster-provision/k8s/1.29/provision.sh

```diff
@@ -0,0 +1,202 @@
+
+function pull_container_retry() {
+    retry=0
+    maxRetries=5
+    retryAfterSeconds=3
+    until [ ${retry} -ge ${maxRetries} ]; do
+        crictl pull "$@" && break
+        retry=$((${retry} + 1))
+        echo "Retrying ${FUNCNAME[0]} [${retry}/${maxRetries}] in ${retryAfterSeconds}(s)"
+        sleep ${retryAfterSeconds}
+    done
+
+    if [ ${retry} -ge ${maxRetries} ]; then
+        echo "${FUNCNAME[0]} Failed after ${maxRetries} attempts!"
+        exit 1
+    fi
+}
+
+# Install modules of the initrd kernel
+dnf install -y "kernel-modules-$(uname -r)"
+
+# Resize root partition
+dnf install -y cloud-utils-growpart
+if growpart /dev/vda 1; then
+    resize2fs /dev/vda1
+fi
+
+dnf install -y patch
+
+systemctl stop firewalld || :
+systemctl disable firewalld || :
+# Make sure the firewall is never enabled again
+# Enabling the firewall destroys the iptable rules
+dnf -y remove firewalld
+
+# Required for iscsi demo to work.
+dnf -y install iscsi-initiator-utils
+
+# required for some sig-network tests
+dnf -y install nftables
+
+# for rook ceph
+dnf -y install lvm2
+# Convince ceph our storage is fast (not a rotational disk)
+echo 'ACTION=="add|change", SUBSYSTEM=="block", KERNEL=="vd[a-z]", ATTR{queue/rotational}="0"' \
+	> /etc/udev/rules.d/60-force-ssd-rotational.rules
+
+# To prevent preflight issue related to tc not found
+dnf install -y iproute-tc
+# Install istioctl
+export PATH="$ISTIO_BIN_DIR:$PATH"
+(
+  set -E
+  mkdir -p "$ISTIO_BIN_DIR"
+  curl "https://storage.googleapis.com/kubevirtci-istioctl-mirror/istio-${ISTIO_VERSION}/bin/istioctl" -o "$ISTIO_BIN_DIR/istioctl"
+  chmod +x "$ISTIO_BIN_DIR/istioctl"
+)
+
+export CRIO_VERSION=1.28
+cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}.repo
+[isv_kubernetes_addons_cri-o_stable_v${CRIO_VERSION}]
+name=CRI-O v${CRIO_VERSION} (Stable) (rpm)
+type=rpm-md
+baseurl=https://storage.googleapis.com/kubevirtci-crio-mirror/isv_kubernetes_addons_cri-o_stable_v${CRIO_VERSION}
+gpgcheck=0
+enabled=1
+EOF
+
+dnf install -y cri-o container-selinux
+
+systemctl enable --now crio
+
+dnf install -y libseccomp-devel
+
+cat << EOF > /etc/containers/registries.conf
+[registries.search]
+registries = ['registry.access.redhat.com', 'registry.fedoraproject.org', 'quay.io', 'docker.io']
+
+[registries.insecure]
+registries = ['registry:5000']
+
+[registries.block]
+registries = []
+EOF
+
+packages_version=$(getKubernetesClosestStableVersion)
+
+# Add Kubernetes release repository.
+# use repodata from GCS bucket, since the release repo might not have it right after the release
+# we deduce the https path from the gcs path gs://kubernetes-release/release/${version}/rpm/x86_64/
+# see https://github.com/kubernetes/kubeadm/blob/main/docs/testing-pre-releases.md#availability-of-pre-compiled-release-artifacts
+cat <<EOF >/etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes Release
+baseurl=https://pkgs.k8s.io/core:/stable:/v1.28/rpm
```

Comment:
This might be the issue with install kubeadm - I shouldn't of hardcoded the version here.

Reply:
Yeah, let's wait until your PR has merged, and rerun the automation.